geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,240 @@
+import org.locationtech.geomesa.web.core.PersistentDataStoreServlet
+import org.scalatra._
+import org.scalatra.servlet.{FileItem, FileUploadSupport, MultipartConfig, SizeConstraintExceededException}
 
 import scala.collection.JavaConversions._
+import scala.collection.{concurrent}
 
-class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
-  override def root: String = "blob"
+class BlobstoreServlet(val persistence: FilePersistence)
+  extends PersistentDataStoreServlet
+    with FileUploadSupport
+    with GZipSupport {
+  override def root: String = "blobstore"
 
-  var abs: AccumuloBlobStore = null
+  val maxFileSize: Int = System.getProperty(BlobstoreServlet.maxFileSizeSysProp, "50").toInt
+  val maxRequestSize: Int = System.getProperty(BlobstoreServlet.maxRequestSizeSysProp, "100").toInt
 
-  // TODO: Revisit configuration and persistence of configuration.
-  // https://geomesa.atlassian.net/browse/GEOMESA-958
-  // https://geomesa.atlassian.net/browse/GEOMESA-984
-  post("/ds") {
+  // caps blob size
+  configureMultipartHandling(
+    MultipartConfig(
+      maxFileSize = Some(maxFileSize * 1024 * 1024),
+      maxRequestSize = Some(maxRequestSize * 1024 * 1024)
+    )
+  )
+  error {
+    case e: SizeConstraintExceededException =>
+      handleError("Uploaded file too large!", e)
+    case e: IOException =>
+      handleError("IO exception in BlobstoreServlet", e)
+  }
 
-    logger.debug("In ds registration method")
+  val blobStores: concurrent.Map[String, AccumuloBlobStore] = new ConcurrentHashMap[String, AccumuloBlobStore]
+  getPersistedDataStores.foreach {
+    case (alias, params) => connectToBlobStore(params).map(abs => blobStores.putIfAbsent(alias, abs))
+  }
 
-    val dsParams = datastoreParams
+  private def connectToBlobStore(dsParams: Map[String, String]): Option[AccumuloBlobStore] = {
     val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+    if (ds == null) {
+      logger.error("Bad Connection Params: {}", dsParams)
+      None
+    } else {
+      Some(new AccumuloBlobStore(ds))
+    }
+  }
+
+  /**
+    * Override of the requestPath function, may need to match on type of request if wrapped
+    * depending on deployment environment
+    *
+    * @param request
+    * @return
+    */
+  override def requestPath(implicit request: HttpServletRequest): String = request match {
+    case r: HttpServletRequestWrapper =>
+      r.getMethod.toLowerCase(Locale.US) match {
+        case "post" =>
+          if (r.getServletPath == "/blob") {
+            s"/blob${r.getPathInfo}"
+          } else {
+            r.getPathInfo
+          }
+        case "delete" =>
+          r.getPathInfo.replace(s"/$root", "")
+        case _        =>
+          r.getPathInfo
+      }
+    case _ => super.requestPath
+  }
+
+  private def handleBadRequest(msg: String): ActionResult = {
+    logger.error(msg)
+    if (debug) {
```

Comment:
don't think we need to hide bad request messages - it's probably helpful to know what you did wrong.